gateway-spi/src/main/java/org/apache/knox/gateway/topology/Provider.java

```diff
@@ -82,4 +84,41 @@ public void setRole( String role ) {
     this.role = role;
   }
 
+  @Override
+  public int hashCode() {
+    return new HashCodeBuilder().append(name)
+                                .append(role)
+                                .append(params)
+                                .append(enabled)
+                                .build();
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
I really like you are using `HashCodeBuilder` above. Why not use `EqualsBuilder` here (this would require to implement `hashCode` and `equals` in `Provider` too)?

Reply:
+1